src/components/table.js

```diff
@@ -905,8 +905,8 @@ export const TooltipCell = ({ children, tooltip, ...props }) =>
     [h(TextCell, props, [children])]
   );
 
-export const HeaderCell = (props) => {
-  return h(TextCell, _.merge({ style: { fontWeight: 600 } }, props));
+export const HeaderCell = ({ children, ...props }) => {
```

Comment:
Is this the fix for the 3rd bullet you mention in the PR description?  This one might worth be splitting out into its own PR with an explanation, as it seems kinda random when placed alongside the rest of the PR :)